public/js/action-list.js

```diff
@@ -76,74 +83,378 @@
+            while (markAsLastActive && this.getDirectionalNext(markAsLastActive, eventKey)) {
+                if (! this.getDirectionalNext(markAsLastActive, eventKey).classList.contains('active')) {
+                    break;
+                }
+
+                markAsLastActive = this.getDirectionalNext(markAsLastActive, eventKey);
+            }
+
+            return [toActiveItem, markAsLastActive];
+        }
+
+        selectAll(list) {
+            let toActive = list.querySelectorAll(':scope > [data-action-item]:not(.active)');
+
+            if (toActive.length) {
+                toActive.forEach(item => item.classList.add('active'));
+                this.setActive(toActive, list.lastChild);
+            }
+        }
+
+        clearSelection(selectedItems) {
+            selectedItems.forEach(item => item.classList.remove('active'));
+        }
+
+        setActive(toActiveItem, markAsLastActive) {
+            if (toActiveItem instanceof HTMLElement) {
+                toActiveItem.classList.add('active');
+
+                toActiveItem.nextElementSibling
+                    ? toActiveItem.nextElementSibling.scrollIntoView({block: "nearest"})
+                    : null;
+                toActiveItem.previousElementSibling
+                    ? toActiveItem.previousElementSibling.scrollIntoView({block: "nearest"})
+                    : null;
+            }
+
+            this.lastActivatedItemUrl = markAsLastActive
+                ? markAsLastActive.dataset.icingaDetailFilter
+                : toActiveItem.dataset.icingaDetailFilter;
+
+            let list = document.querySelector('.action-list');
+            let activeItems = list.querySelectorAll(':scope > [data-action-item].active');
+
+            if (activeItems.length > 1) {
+                this.detailUrl = this.createMultiSelectUrl(activeItems);
+            } else {
+                this.detailUrl = toActiveItem.querySelector('[href]').getAttribute('href');
+            }
+
+            this.addSelectionCountToFooter(list);
+
+            clearTimeout(this.lastTimeoutId);
+            this.lastTimeoutId = setTimeout(() => {
+                this.icinga.loader.loadUrl(this.detailUrl, this.icinga.loader.getLinkTargetFor($(toActiveItem)));
+                delete document.querySelector('#col1').dataset.suspendAutorefresh;
+            }, 250);
+        }
+
+        handleLoadMoreNavigate(loadMoreElement, lastActivatedItem) {
+            let req = this.loadMore($(loadMoreElement.querySelector('a')));
+            this.isProcessingLoadMore = true;
+            req.done(() => {
+                this.isProcessingLoadMore = false;
+                // list has now new items, so select the lastActivatedItem and then move forward
+                let toActiveItem = document.querySelector('.action-list').querySelector(`[data-icinga-detail-filter="${ lastActivatedItem.dataset.icingaDetailFilter }"]`);
```

Comment:
a) Don't use `document` as reference. Use `loadMoreElement.closest('.action-list')` instead.
b) Wrap long lines also in JS please